### PR TITLE
Harvester Cleanup and FileEvent

### PR DESCRIPTION
**Pull request by ruflin**
* Rename FileEvent to Event
* Remove double information between FileEvent and State (Source, Fileinfo)
* Remove readLine function as not needed anymore

### CODE REVIEW

File: filebeat/harvester/log.go [hunk 1]
```diff
@@ -63,8 +64,7 @@ func (h *Harvester) Harvest() {
 		default:
 		}
 
-		// Partial lines return error and are only read on completion
-		ts, text, bytesRead, jsonFields, err := readLine(r)
+		message, err := r.Next()
```
Comment (ruflin) [inline filebeat/harvester/log.go]: @urso The behaviour here should be identical to before, but hopefully better readable. An event is still created if Byte == 0 but this is filtered out as a state update only by the publisher. I assume this happens when an event only consists of a newline. So the state will be updated no event will be sent to the output. Alternatively we could also check inside shouldExportLine for bytes == 0.

Comment (urso) [inline filebeat/harvester/log.go]: normally new lines characters are added to byte count, so to have correct offset.

Comment (urso) [inline filebeat/harvester/log.go]: +1 for using struct over 4 variables

**Comment by urso** [comment]
LGTM. waiting for travis.

**Pull request merged**